### DISCUSSION TITLE
[12 settembre 2024] Esercizio 1

### DISCUSSION THREAD

**Discussion by Nw4pS** [sapienzastudentsnetwork/psd2223#269] [Esami]
Progettare un circuito sequenziale con due ingressi x1, x0, che codificano i caratteri A, G, S nel seguente modo: 

<div align="center">

<table>
  <tr>
    <th align="center">$\Large x1, x0$</th>
    <th align="center">carattere</th>
  </tr>
  <tr>
    <td align="center">$\Large 00$</td>
    <td align="center">$\Large A$</td>
  </tr>
  <tr>
    <td align="center">$\Large 01$</td>
    <td align="center">$\Large G$</td>
  </tr>
  <tr>
    <td align="center">$\Large 10$</td>
    <td align="center">$\Large S$</td>
  </tr>
</table>

</div>

Il circuito ha 2 uscite z1 e z0 e fornisce z1=1 quando riceve in ingresso la sequenza GAS e z0=1 quando riceve in ingresso la sequenza GAG. Sono ammesse sovrapposizioni. 

<details closed>
    <summary> soluzione A-L </summary>

<div align="center">

<img src="https://github.com/user-attachments/assets/5993e8ea-f957-422a-9d1c-38696622e541">

![image](https://github.com/user-attachments/assets/108c03f6-4a35-4e2b-9915-c9826bee2e49)

</div>
</details>

**Comment by FeddyLix17** [comment]
<div align="center">

<table>
<tr>
<td>
<table>
    <tr>
        <th>Stati</th>
        <th>$\Large S_1$</th>
        <th>$\Large S_0$</th>
    </tr>
    <tr>
        <th>$\Large S0/R$</th>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <th>$\Large S1/G$</th>
        <td align="center">0</td>
        <td align="center">1</td>
    </tr>
    <tr>
        <th>$\Large S2/GA$</th>
        <td align="center">1</td>
        <td align="center">0</td>
    </tr>
</table>
</td>
<td>
<img src="https://github.com/user-attachments/assets/7ca44ede-e532-4633-aae7-0257b2fdcf51" width="100%" height="50%">
</td>
</tr>
</table>

<table>
    <tr>
        <th>PS</th>
        <th rowspan="22"> </th>
        <th>$\Large S_1$</th>
        <th>$\Large S_0$</th>
        <th rowspan="22"> </th>
        <th>$\Large x1$</th>
        <th>$\Large x0$</th>
        <th rowspan="22"> </th>
        <th>NS</th>
        <th rowspan="22"> </th>
        <th>$\Large S_1'$</th>
        <th>$\Large S_0'$</th>
        <th rowspan="22"> </th>
        <th>$\Large z1$</th>
        <th>$\Large z0$</th>
    </tr>
    <tr>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">G</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">G</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">GA</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">G</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">G</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">G</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">GA</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">GA</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">G</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">1</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">GA</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
    </tr>
</table>

</div>

$$
    \Large z1 = S_1\overline{S_0}x1\overline{x0} \quad \quad z0 = S_1\overline{S_0}\overline{x1}x0
$$

$$
    \Large S_1' = \overline{S_1}S_0\overline{x1}\overline{x0}
$$

$$
    \Large S_0' = \overline{S_1}\overline{S_0}\overline{x1}x0 + \overline{S_1}S_0\overline{x1}x0 + S_1\overline{S_0}\overline{x1}x0 =
$$

$$
    \Large = \overline{S_1}\overline{x1}x0(S_0 + \overline{S_0}) + S_1\overline{S_0}\overline{x1}x0 =
$$

$$
    \Large = \overline{S_1}\overline{x1}x0 + S_1\overline{S_0}\overline{x1}x0 =
$$

$$
    \Large = \overline{x1}x0(\overline{S_1} + S_1\overline{S_0}) =
$$

$$
    \Large = \boxed{\overline{x1}x0(\overline{S_1} + \overline{S_0})}
$$

> [!NOTE]
> Le equazioni di $\Large S_1'$ e $\Large S_0'$ sono diverse da quelle della soluzione ufficiale (trovo che siano comunque corrette, fatemi sapere se trovate qualche typo)

![image](https://github.com/user-attachments/assets/da9a91fe-4d5c-4e12-9fb7-70b73f8bd096)

**Reply by markcst** [reply]
Ciao, anche a me escono le tue stesse equazioni, ho fatto con le k-map.

Come mai al prof escono quelle? Sono o non sono equivalenti alle nostre, eventualmente?

**Reply by iBiond** [reply]
credo che abbia semplificato le espressioni utilizzando lo stato e l'input 11 come don't care, non capisco però perché l'abbia fatto con gli stati futuri e non con le uscite
